internal/api/api.go

```diff
@@ -25,40 +26,48 @@ type snykApiClient struct {
 }
 
 func (a *snykApiClient) GetOrgIdFromSlug(slugName string) (string, error) {
-	url := a.url + "/v1/orgs"
+	url := GetOrgsApiURL(a.url, slugName)
+
 	res, err := a.client.Get(url)
 	if err != nil {
 		return "", err
 	}
-	defer res.Body.Close()
+	defer func() { _ = res.Body.Close() }()
 
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
 		return "", err
 	}
 
-	var userOrgInfo contract.OrganizationsResponse
-	err = json.Unmarshal(body, &userOrgInfo)
+	var response contract.OrganizationsResponse
+	err = json.Unmarshal(body, &response)
 	if err != nil {
 		return "", err
 	}
 
-	for _, org := range userOrgInfo.Organizations {
-		if org.Slug == slugName {
-			return org.ID, nil
+	organizations := response.Organizations
+	for _, organization := range organizations {
+		if organization.Attributes.Slug == slugName {
+			return organization.Id, nil
 		}
 	}
-
 	return "", fmt.Errorf("org ID not found for slug %v", slugName)
 }
 
+func GetOrgsApiURL(baseURL, slugName string) string {
+	apiVersion := "version=2024-03-12"
+	escapedSlug := url2.PathEscape(slugName)
+	url := fmt.Sprintf("%s/rest/orgs?%s&slug=%s", baseURL, apiVersion, escapedSlug)
+	return url
+}
```

Comment:
Suggest constructing a `*url.URL` instead, and using the API to set the query parameters, letting the stdlib deal with escaping. Something like this:

```suggestion
func OrgsApiURL(baseURL, slugName string) (*url.URL, error) {
	u, err := url2.Parse(baseURL+"/rest/orgs")
	if err != nil {
		return nil, err
	}
	q := u.Query()
	q.Set("version", orgsAPIVersion)  // use a constant for the orgs API
	q.Set("slug", slugName)
	return u, nil
}
```

Other variations, you could accept `baseURL` as a *url.URL, copy the struct and append the path (avoiding the need for a parse).

I'd also drop the `Get` prefix (https://go.dev/doc/effective_go#Getters). `New` might be ok if it's creating a *url.URL.

Reply:
Done. Only added the Get prefix, because it's rather common in go-application-framework, even though it's not golang-style. But yeah, no worries, dropped it.